apollo-federation/src/operation/mod.rs

```diff
@@ -596,11 +596,21 @@ pub(crate) enum Selection {
     InlineFragment(Arc<InlineFragmentSelection>),
 }
 
+/// Element enum that is more general than OpPathElement.
+/// - Used for operation optimization.
+#[derive(Debug, Clone, derive_more::From)]
+pub(crate) enum OperationElement {
+    Field(Field),
+    FragmentSpread(FragmentSpread),
+    InlineFragment(InlineFragment),
+}
```

Comment:
I kinda recoiled at first seeing yet another selection type, but now we have this set of data structures:

| Data | Cached key wrapper | Selection wrapper |
|------|----------------|-|
| FieldData | Field | FieldSelection |
| InlineFragmentData | InlineFragment | InlineFragmentSelection |
| FragmentSpreadData | FragmentSpread | FragmentSpreadSelection |
| | OperationElement | Selection |

(plus like `OpPathElement` which only somewhat qualifies, and doesnt fit in the table)

I think that's a pretty acceptable place to be because we can, I think, generalise the middle and right columns into a single ergonomic generic type each. I wrote some notes on it in FED-302.

Reply:
I know. That's the situation for now. I'll follow up on FED-302.